### PR TITLE
New pattern - REST API hello world (java)

### PR DESCRIPTION
**Pull request by seshubaws**
*Issue #, if available:*

*Description of changes:*

Added template for REST API + Lambda Hello World tutorial in Java21 for the getting started project (guided walkthroughs). This tutorial is from the SAM CLI app templates: https://github.com/aws/aws-sam-cli-app-templates/tree/master/java21/hello-maven

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apigw-rest-api-lambda-java/HelloWorldFunction/src/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+      <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-shade-plugin</artifactId>
+          <version>3.2.4</version>
```
Comment (bfreiberg) [inline apigw-rest-api-lambda-java/HelloWorldFunction/src/pom.xml:38]: Any reason not to use the latest version, 3.6.0?

File: apigw-rest-api-lambda-java/HelloWorldFunction/src/pom.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    <properties>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-lambda-java-core</artifactId>
+            <version>1.2.2</version>
```
Comment (bfreiberg) [inline apigw-rest-api-lambda-java/HelloWorldFunction/src/pom.xml:18]: Any reason not to use the latest version, 1.2.3?

File: apigw-rest-api-lambda-java/example.pattern.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "cleanup": {
+      "text": [
+        "Delete the application: <code>sam delete</code>."
+      ]
+    },
+    "authors": [
+    {
+      "name": "Seshu Brahma",
+      "image": "https://i.postimg.cc/ZR0MyrTN/seshub-photo.jpg",
+      "bio": "SDE II, Lambda, AWS",
```
Comment (bfreiberg) [inline apigw-rest-api-lambda-java/example.pattern.json:53]: see my comment on Slack, can we change this to something recognizable to an external user?

File: apigw-rest-api-lambda-java/HelloWorldFunction/src/pom.xml (hunk 2 above)
Comment (seshubaws) [reply apigw-rest-api-lambda-java/HelloWorldFunction/src/pom.xml]: Thanks for catching, updated!

**Comment by bfreiberg** [comment]
Looks good, thank you for your contribution. Your PR will be merged to Serverlessland.com soon

**Pull request merged**